base/ca/src/main/java/com/netscape/ca/CAService.java

```diff
@@ -455,11 +451,10 @@ public boolean serviceRequest(IRequest request) {
                                 message));
 
                         return true;
-                    } else {
-                        if (request.getExtDataInString(IRequest.ERROR) != null) {
-                            request.setExtData(IRequest.RESULT, IRequest.RES_SUCCESS);
-                            request.deleteExtData(IRequest.ERROR);
-                        }
+                    }
+                    if (request.getExtDataInString(IRequest.ERROR) != null) {
+                        request.setExtData(IRequest.RESULT, IRequest.RES_SUCCESS);
+                        request.deleteExtData(IRequest.ERROR);
```

Comment:
We might not want to execute this if the preceding `if` was executed

Could use `else if` instead

Reply:
By preceding `if` do you mean the one on new line 440? If so that `if` block contains a `return` statement so there is no need for the `else` clause.

Reply:
Ah handy. I didn't notice that.